src/test/app/Offer_test.cpp

```diff
@@ -4122,13 +4206,187 @@ class Offer_test : public beast::unit_test::suite
         env (offer (alice, gwUSD(40), XRP(4000)));
         env.close();
 
+        env.require (offers (alice, 1));
+
+        // Now bob creates his offer again.  alice's offer should cross.
+        env (offer (bob, XRP(4000), gwUSD(40)));
+        env.close();
+
         env.require (offers (alice, 0));
         env.require (balance (alice, gwUSD(40)));
 
         env.require (offers (bob, 0));
         env.require (balance (bob, gwUSD(10)));
     }
 
+    void testRCSmoketest(std::initializer_list<uint256> fs)
+    {
+        testcase("RippleConnect Smoketest payment flow");
+        using namespace jtx;
+
+        Env env {*this, fs};
+        auto const closeTime =
+            fix1449Time() +
+                100 * env.closed()->info().closeTimeResolution;
+        env.close (closeTime);
+
+        // This test mimics the payment flow used in the Ripple Connect
+        // smoke test.  The players:
+        //   A USD gateway with hot and cold wallets
+        //   A EUR gateway with hot and cold walllets
+        //   A MM gateway that will provide offers from USD->EUR and EUR->USD
+        // A path from hot US to cold EUR is found and then used to send
+        // USD for EUR that goes through the market maker
+
+        auto const hotUS = Account("hotUS");
+        auto const coldUS = Account("coldUS");
+        auto const hotEU = Account("hotEU");
+        auto const coldEU = Account("coldEU");
+        auto const mm = Account("mm");
+
+        auto const USD = coldUS["USD"];
+        auto const EUR = coldEU["EUR"];
+
+        env.fund (XRP(100000), hotUS, coldUS, hotEU, coldEU, mm);
+        env.close();
+
+        // Cold wallets require trust but will ripple by default
+        for( auto & cold : {coldUS, coldEU})
```

Comment:
Nit: formatting. Might as well throw a `const` on `cold` while you're changing the formatting. I.e:
```for(auto const& cold : {coldUS, coldEU})```